### ISSUE TITLE
Unable to remove temporary SMTP address of group (original address not added)

### ISSUE THREAD

**Issue by Arne-RFA** [timmcmic/DLConversionV2#176]
After setting up two DL to test with, and synced them to 365 ran the multiple migration and it all looked good until the below (same error in both tests)

```
[6/6/2024 12:22:12 PM] - =====
[6/6/2024 12:22:12 PM] - Post Create Errors:
[6/6/2024 12:22:12 PM] - Primary Email Address or UPN: zzzDLtest1@clientpublicdomain.com
[6/6/2024 12:22:12 PM] - External Directory Object ID: ed02fdad-7d89-4917-b29b-5c8ae144d686
[6/6/2024 12:22:12 PM] - Name: zzzDLtest1
[6/6/2024 12:22:12 PM] - Alias: zzzDLtest1
[6/6/2024 12:22:12 PM] - Attribute in Error: Unable to remove temporary SMTP address of group.
[6/6/2024 12:22:12 PM] - Error Message: Unable to remove20240606T12095141101728395316@clientpublicdomainTenant.onmicrosoft.com - manual removal required.
[6/6/2024 12:22:12 PM] - Error Message Details: ExC71195|Microsoft.Exchange.Data.DataValidationException|There is no primary SMTP address.
[6/6/2024 12:22:12 PM] - =====
[6/6/2024 12:22:12 PM] - =====
[6/6/2024 12:22:12 PM] - General Errors:
[6/6/2024 12:22:12 PM] - Error Message: Unable to locate a mail.onmicrosoft.com address either on premises or office 365.  This may indicated larger failures to set SMTP addresses.  Manual correction required and run enable-hybridMailFlowPostMigration to fix.
[6/6/2024 12:22:12 PM] - Error Message Detail: 
[6/6/2024 12:22:12 PM] - =====

```
The AADC Synced DL have been "replaced" with cloud only DL, but only have the "temporary" SMTP addresses on them

Within AD it shows as 

![image](https://github.com/timmcmic/DLConversionV2/assets/74773674/d7e6e6e0-d865-431f-8f13-93947e8ac983)

Happy to send over the full logs, but cannot attach them here.

**Comment by timmcmic** [comment]
Right - to the on premises email address policy should be setting a mail.Microsoft.com address.  When you use enable hybrid mail flow this is what we use for the target address.  If there is not one then this portion errors, you need to add one, and enable hybrid mail flow after.

I thought I made a change to just randomize this - I’ll need to go back and check.


==============================
Timothy J. McMichael
Senior Support Escalation Engineer
***@***.******@***.***>
(980)-776-7465

Hours:  Sunday – Wednesday 08:00 – 16:00 eastern time zone.

Manager:  Tom Roughley ***@***.******@***.***>)

Premier Support - (800)-936-3100
Broad Commercial Support - (800)-936-4900

==============================

________________________________
From: Arne-RFA ***@***.***>
Sent: Thursday, June 6, 2024 9:39 AM
To: timmcmic/DLConversionV2 ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [timmcmic/DLConversionV2] Unable to remove temporary SMTP address of group (original address not added) (Issue #176)


After setting up two DL to test with, and synced them to 365 ran the multiple migration and it all looked good until the below (same error in both tests)

[6/6/2024 12:22:12 PM] - =====
[6/6/2024 12:22:12 PM] - Post Create Errors:
[6/6/2024 12:22:12 PM] - Primary Email Address or UPN: ***@***.***
[6/6/2024 12:22:12 PM] - External Directory Object ID: ed02fdad-7d89-4917-b29b-5c8ae144d686
[6/6/2024 12:22:12 PM] - Name: zzzDLtest1
[6/6/2024 12:22:12 PM] - Alias: zzzDLtest1
[6/6/2024 12:22:12 PM] - Attribute in Error: Unable to remove temporary SMTP address of group.
[6/6/2024 12:22:12 PM] - Error Message: Unable to ***@***.*** - manual removal required.
[6/6/2024 12:22:12 PM] - Error Message Details: ExC71195|Microsoft.Exchange.Data.DataValidationException|There is no primary SMTP address.
[6/6/2024 12:22:12 PM] - =====
[6/6/2024 12:22:12 PM] - =====
[6/6/2024 12:22:12 PM] - General Errors:
[6/6/2024 12:22:12 PM] - Error Message: Unable to locate a mail.onmicrosoft.com address either on premises or office 365.  This may indicated larger failures to set SMTP addresses.  Manual correction required and run enable-hybridMailFlowPostMigration to fix.
[6/6/2024 12:22:12 PM] - Error Message Detail:
[6/6/2024 12:22:12 PM] - =====



The AADC Synced DL have been "replaced" with cloud only DL, but only have the "temporary" SMTP addresses on them

Within AD it shows as

image.png (view on web)<https://github.com/timmcmic/DLConversionV2/assets/74773674/d7e6e6e0-d865-431f-8f13-93947e8ac983>

Happy to send over the full logs, but cannot attach them here.

**Comment by Arne-RFA** [comment]
Each DL has the primary SMTP address which is the public domain (added by address policy) and the tenant.mail.onmicrosoft.com proxy address (added by policy)

The next two test DLs I have setup I have confirmed have the tenantdomain.mail.onmicrosoft.com proxy addresses, I will test those first thing tomorrow morning (UK timezone) and see if the same issue repeats.

Very happy to send logs to an @Microsoft address, but have no desire to send you anything unnecessary (-:

**Comment by Arne-RFA** [comment]
Confirmed in originalDLConfigurationADXML.xml that the original testDL **did** have a mail.onmicrosoft.com address

![image](https://github.com/timmcmic/DLConversionV2/assets/74773674/0894699b-edb6-4184-b867-cdf3afdd6933)

Same test now running with two more test DL.

**Comment by timmcmic** [comment]
I'm not sure what you're saying at this point - are you saying that the error is false and the DL did have this address when you migrated.

If so you can send your logs to dlconversionv2 at service dot Microsoft dot com

Tim

**Comment by Arne-RFA** [comment]
Apologies for my lack of clarity, I am saying that the DL had a public domain SMTP address and a mail.onmicrosoft.com smtp address before the migration, so the failure of the script was "unexpected".

I'll zip up and mail the logs over (-:

**Comment by timmcmic** [comment]
Yes - that would be unexpected.

**Comment by Arne-RFA** [comment]
Closed with 2.9.8.47 (-:

**Comment by Arne-RFA** [comment]
Closed with 2.9.8.47 (-:
